resources/smpp/ra/src/main/java/org/restcomm/slee/resource/smpp/SmppSessionsImpl.java

```diff
@@ -180,16 +160,33 @@ private void countSendResponsePdu(SmppSessionCounters counters, PduResponse pdu,
             // TODO: adding here statistics for SUBMIT_MULTI ?
             }
         }
-    }	
+    }
 
-	protected class SmppSessionHandlerInterfaceImpl implements SmppSessionHandlerInterface {
+	protected class SmppSessionHandlerInterfaceImpl implements
+			SmppSessionHandlerInterface {
 
 		public SmppSessionHandlerInterfaceImpl() {
 
 		}
 
+		public void destroySmppSessionHandler(Esme esme) {
+			if (esme != null && esme.getName() != null) {
+				SenderThread senderThread = esmeSenderThreads.remove(esme
+						.getName());
+				if (senderThread != null)
+					senderThread.deactivate();
+			}
+		}
+
 		@Override
 		public SmppSessionHandler createNewSmppSessionHandler(Esme esme) {
+
+			SenderThread senderThread = new SenderThread("SMPP ESME Sender " + esme.getName());
+			senderThread.start();
```

Comment:
why dont we simply start the thread only if putIfAbnset returns "null", instead of starting threads that are not goign to be used?